dataset_interfaces/factory.py

```diff
@@ -60,3 +62,33 @@ def create_examples(dataset_config: dict, universal_args: dict, max_message_size
             if example.example_id == "":
                 example.example_id = str(i)
         return examples
+
+    @staticmethod
+    def create_dataset_for_example(run_configuration: dict,  test_example_path: str) -> DatasetInterface:
+        args = deepcopy(run_configuration["datasets"]["args"])
+
+        # Get the name of the dataset
+
+        path_dataset_name = test_example_path.split(os.sep)[-2]
```

Comment:
It would be better to use `utils.files.parse_result_path`

Reply:
I've just noticed it's only for result files, but we could have one for example files too.